### PR TITLE
Simple variable equation term refactoring

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java [hunk 1]
```diff
@@ -103,6 +106,76 @@ static EquationTerm multiply(EquationTerm term, double scalar) {
         return new MultiplyByScalarEquationTerm(term, scalar);
     }
 
+    class VariableEquationTerm extends AbstractEquationTerm {
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:109]: Ok, but in that case do we let the static factory in EquationTerm on move it in VariableEquationTerm? 
Like:
```java
VariableEquationTerm.create(...)
```

[the remark this answers is not in the document]
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:109]: Maybe we shoud also also move it I think?

[the remark this answers is not in the document]
Comment (floriand-e2r) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:109]: ah, I liked the `EquationTerm.createVariable()`... but yes, indeed, we also need to move it to break that dependency

[the remark this answers is not in the document]
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:109]: We have to choose: 
either like in this PR 
```java
EquationTerm.createVariable(...)
```
Or
```java
VariableEquationTerm.create(...)
```

So I change the code, or I keep it as it is?

[the remark this answers is not in the document]
Comment (floriand-e2r) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:109]: I vote for `EquationTerm.createVariable(...)`

[the remark this answers is not in the document]
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:109]: So you vote for keeping the PR as it is?

[the remark this answers is not in the document]
Comment (floriand-e2r) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:109]: yes, sorry, that implies keeping the PR as it is. I withdraw my comment!

**Review by floriand-e2r — APPROVED** [review]

File: src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java [hunk 2]
```diff
@@ -103,6 +106,76 @@ static EquationTerm multiply(EquationTerm term, double scalar) {
...
+            return 0;
+        }
+
+        @Override
+        public void write(Writer writer) throws IOException {
+            variables.get(0).write(writer);
+        }
+    }
+
+    static VariableEquationTerm createVariable(LfElement element, VariableType variableType, VariableSet variableSet) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:168]: The name of the method is really confusing for me. I think that we have to be totally explicit: `createVariableEquationTerm`?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:168]:
```java
EquationTerm.createVariableEquationTerm
```
would be redundant ?

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java:168]: `createVariableTerm`?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/equations/EquationTerm.java]: Done!

**Review by floriand-e2r — APPROVED** [review]

**Pull request merged**